### DISCUSSION TITLE
eKuiper v2 brain storming

### DISCUSSION THREAD

**Discussion by ngjaying** [lf-edge/ekuiper#1867] [Ideas]
We are excited to announce the launch of eKuiper version 2, a major update to our open source edge streaming SQL engine. With the major version update opportunity, we can refactor the design, solve technical debt and bring new features and improvements that make eKuiper more powerful, flexible and user-friendly. 

We are now collecting ideas for v2.  We welcome your feedback and ideas on how to make eKuiper even better for your use cases. Please add to this thread if you have any ideas. Thank you!

**Comment by canob** [comment]
Hi everyone,
One idea that I didn't see even on bigger stream processing products, is the idea of do anomaly detection over logs/metrics on a simple way, and in this case, on the edge.
The advantage of eKuiper in this case is that eKuiper can even create the metrics for different kind of logs by itself with rules if the original technology don't provide them.
For example, first you can create a rule to generate metrics from firewall logs for denied connections by IP/user/app (depends on if you are using a common firewall or a NG firewall), that aggregate that over a minute, and after that, you can use a algorithm like Random Cat Forest (RCF), https://docs.aws.amazon.com/sagemaker/latest/dg/randomcutforest.html, which is an unsupervised algorithm for detecting anomalous data points within a data set. 
One example that show that you can implement this on the edge or as stream processing option is Data Prepper Anomaly Detector Processor, from OpenSearch, https://opensearch.org/docs/2.11/data-prepper/pipelines/configuration/processors/anomaly-detector/.
For me, this can be a big differential for eKuiper over other data stream processing engines.
